src/sentry/static/sentry/app/components/alertLink.tsx

```diff
@@ -1,29 +1,34 @@
 import styled from 'react-emotion';
-import PropTypes from 'prop-types';
 import React from 'react';
+import {LocationDescriptor} from 'history';
 
 import Link from 'app/components/links/link';
 import InlineSvg from 'app/components/inlineSvg';
 import space from 'app/styles/space';
 
-export default class AlertLink extends React.Component {
-  static propTypes = {
-    to: PropTypes.string,
-    href: PropTypes.string,
-    icon: PropTypes.string,
-    priority: PropTypes.oneOf(['info', 'warning', 'success', 'error', 'muted']),
-    size: PropTypes.oneOf(['small', 'normal']),
-  };
+type Size = 'small' | 'normal';
+type Priority = 'info' | 'warning' | 'success' | 'error' | 'muted';
+
+type Props = {
+  // do we need to enforce here to always provide either href or to?
+  href?: string;
+  to?: LocationDescriptor;
+  icon?: string;
+  size: Size;
+  priority: Priority;
+  onClick?: (e: React.MouseEvent) => void;
+};
 
-  static defaultProps = {
+export default class AlertLink extends React.Component<Props> {
+  static defaultProps: Partial<Props> = {
     priority: 'warning',
+    size: 'normal',
   };
 
   render() {
-    const {icon, children, size} = this.props;
-
+    const {to, href, size, priority, icon, children, onClick} = this.props;
     return (
-      <StyledLink {...this.props}>
+      <StyledLink {...{to, href, size, priority, onClick}}>
```

Comment:
Was there an error created by `{...this.props}`? 

This approach looks weird because you're creating an object and destructuring it immediately. It might be better to just use `<StyledLink to={to} href={href} ...etc>` instead.

Reply:
I didn't want to pass all props to Link, because it would create a problem once the Link itself is converted to TS - but I agree that cherrypicking props like that looks weird, I changed it to the way you recommended. Thanks.